From 3182e9dce84112265a9ead1f8e346d64e497979e Mon Sep 17 00:00:00 2001
From: Oliver Smith <ollieparanoid@postmarketos.org>
Date: Sat, 14 Sep 2019 01:52:28 +0200
Subject: [PATCH] Add missing throw_exception.hpp include

Fix these errors with boost 1.71.0:

/home/pmos/build/src/mir-1.1.2/src/server/report/lttng/lttng_report_factory.cpp:67:5:
error: use of undeclared identifier 'BOOST_THROW_EXCEPTION'
    BOOST_THROW_EXCEPTION(std::logic_error("Not implemented"));
    ^
/home/pmos/build/src/mir-1.1.2/src/server/report/lttng/lttng_report_factory.cpp:77:5:
error: use of undeclared identifier 'BOOST_THROW_EXCEPTION'
    BOOST_THROW_EXCEPTION(std::logic_error("Not implemented"));
---
 src/server/report/lttng/lttng_report_factory.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/server/report/lttng/lttng_report_factory.cpp b/src/server/report/lttng/lttng_report_factory.cpp
index d4a98b3b65..55ae6d7b65 100644
--- a/src/server/report/lttng/lttng_report_factory.cpp
+++ b/src/server/report/lttng/lttng_report_factory.cpp
@@ -26,6 +26,7 @@
 #include "scene_report.h"
 #include "session_mediator_report.h"
 #include "shared_library_prober_report.h"
+#include <boost/throw_exception.hpp>
 
 std::shared_ptr<mir::compositor::CompositorReport> mir::report::LttngReportFactory::create_compositor_report()
 {
-- 
2.23.0

